3d_generation/maisi/scripts/infer_using_diff_model.py

```diff
@@ -0,0 +1,586 @@
+            bottom_region_index_tensor = np.array([0, 0, 1, 0]).astype(float)
+            # bottom_region_index_tensor = np.array([0, 0, 0, 1]).astype(float)
+
+            # out_spacing = [1.0, 1.0, 1.0]
+            spacing_tensor = np.array(out_spacing).astype(float)
+
+            top_region_index_tensor = top_region_index_tensor * 1e2
+            print(f"top_region_index_tensor: {top_region_index_tensor}.")
+            bottom_region_index_tensor = bottom_region_index_tensor * 1e2
+            print(f"bottom_region_index_tensor: {bottom_region_index_tensor}.")
+            spacing_tensor = spacing_tensor * 1e2
+            print(f"spacing_tensor: {spacing_tensor}.")
+
+            top_region_index_tensor = top_region_index_tensor[np.newaxis, :]
+            bottom_region_index_tensor = bottom_region_index_tensor[np.newaxis, :]
+            spacing_tensor = spacing_tensor[np.newaxis, :]
+
+            if amp:
+                top_region_index_tensor = torch.from_numpy(top_region_index_tensor).half().to(device)
+                bottom_region_index_tensor = torch.from_numpy(bottom_region_index_tensor).half().to(device)
+                spacing_tensor = torch.from_numpy(spacing_tensor).half().to(device)
+            else:
+                top_region_index_tensor = torch.from_numpy(top_region_index_tensor).float().to(device)
+                bottom_region_index_tensor = torch.from_numpy(bottom_region_index_tensor).float().to(device)
+                spacing_tensor = torch.from_numpy(spacing_tensor).float().to(device)
+
+            outputs = DiffusionInferer.sample(
+                inferer,
+                input_noise=noise,
+                diffusion_model=unet,
+                scheduler=scheduler,
+                save_intermediates=False,
+                intermediate_steps=False,
+                top_region_index_tensor=top_region_index_tensor,
+                bottom_region_index_tensor=bottom_region_index_tensor,
+                spacing_tensor=spacing_tensor,
+            )
+            print("outputs", outputs.size())
+
+            # synthetic_images = autoencoder.decode_stage_2_outputs(outputs / scale_factor)
+            # print("synthetic_images:", synthetic_images.size(), synthetic_images.device)
+
+        with torch.cuda.amp.autocast(enabled=True):
+            target_shape = output_size
+            recon_pt_nda = torch.zeros(
+                (1, 1, target_shape[0], target_shape[1], target_shape[2]),
+                dtype=outputs.dtype,
+            ).to("cuda")
+            _count = torch.zeros(
+                (1, 1, target_shape[0], target_shape[1], target_shape[2]),
+                dtype=outputs.dtype,
+            ).to("cuda")
+            z = outputs
+
+            _temp = recon1(
+                z[
+                    ...,
+                    : target_shape[0] // 4 // 4 * 3,
+                    : target_shape[1] // 4 // 4 * 3,
+                    : target_shape[2] // 4 // 4 * 3,
+                ],
```

Comment:
```suggestion
                z[
                    ...,
                    : target_shape[0] // 4 // 4 * 3,
                    : target_shape[1] // 4 // 4 * 3,
                    : target_shape[2] // 4 // 4 * 3
                ],
```

All of these strange formattings are caused by superfluous commas. I've seen this elsewhere and it's an odd thing black does.

Reply:
Hi @ericspod, I have updated the script. Please proceed. Thanks.

Reply:
Closing this for now as it is no longer valid